popcorn-cli: init at 1.3.30

ethanthoma opened the pull request:
https://github.com/gpu-mode/popcorn-cli

a command-line interface tool for submitting solutions to the Popcorn Discord Bot from gpu-mode.

## Things done:
- Built on platform:
    - [x] x86_64-linux
    - [ ] aarch64-linux
    - [ ] x86_64-darwin
    - [ ] aarch64-darwin
- Tested, as applicable:
    - [ ] [NixOS tests](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) in [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests).
    - [ ] [Package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests) at passthru.tests.
    - [ ] Tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test) for functions and "core" functionality.
- [ ] Ran nixpkgs-review on this PR. See [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage).
- [x] Tested basic functionality of all binary files, usually in ./result/bin/.
- Nixpkgs Release Notes
    - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
    - [ ] Module addition: when adding a new NixOS module.
    - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other READMEs.
---
Add a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to [pull requests you find important](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

ethanthoma committed:
maintainers: add ethanthoma

yzhou216 commented on pkgs/by-name/po/popcorn-cli/package.nix:
```diff
@@ -0,0 +1,43 @@
+in
+rustPlatform.buildRustPackage {
+  pname = "popcorn-cli";
+  inherit version;
```
```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
  pname = "popcorn-cli";
  version = "1.1.44";
```

ethanthoma commented:
Updated to latest release v1.1.46. The version mismatch issue persists (binary still reports 0.1.0), but I've opened https://github.com/gpu-mode/popcorn-cli/issues/25 to track it upstream.

Changes made based on review feedback:
- Refactored to use `finalAttrs` pattern
- Changed `rev` to `tag` parameter
- Added upstream issue link to comment
- Updated to latest version (1.1.46)

yzhou216 reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

FliegendeWurst commented on pkgs/by-name/po/popcorn-cli/package.nix:
```diff
@@ -0,0 +1,41 @@
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  # Version check disabled: binary reports 0.1.0 while git tag is v1.1.46
```
This was fixed in the latest release, but you need to set `env.CLI_VERSION = finalAttrs.version;`

ethanthoma replied:
done, set env.CLI_VERSION = finalAttrs.version and re-enabled versionCheckHook. binary now reports the correct version.

ethanthoma committed:
popcorn-cli: init at 1.3.30

ethanthoma committed:
popcorn-cli: init at 1.3.30

ethanthoma commented:
rebased on master and updated to latest release v1.3.30.

- set `env.CLI_VERSION = finalAttrs.version` per review (upstream fixed https://github.com/gpu-mode/popcorn-cli/issues/25)
- re-enabled `versionCheckHook`, binary now reports 1.3.30
- dropped `openssl`/`pkg-config`: reqwest uses rustls-tls as of this release, no openssl in the lockfile

built locally on x86_64-linux, all 50 upstream tests pass and the version check succeeds.
